presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/LogFilter.java

```diff
@@ -43,28 +43,53 @@
 {
     private final Path file;
     private final Pattern logPattern;
+    private final String defaultEntry;
     private final String lineSeparator;
     private final ImmutableMap<String, Predicate<String>> namedGroupFilters;
     private final ImmutableMap<Integer, Predicate<String>> numberedGroupFilters;
     private final int maxEntries;
     private final boolean keepFirst;
 
-    private LogFilter(Path file, Pattern logPattern, String lineSeparator,
+    private LogFilter(Path file, Pattern logPattern,
+            String defaultEntry, String lineSeparator,
             Map<String, Predicate<String>> namedGroupFilters,
             Map<Integer, Predicate<String>> numberedGroupFilters,
             int maxEntries, boolean keepFirst)
             throws FileNotFoundException
     {
         this.file = requireNonNull(file);
         this.logPattern = requireNonNull(logPattern);
+        this.defaultEntry = requireNonNull(defaultEntry);
         this.lineSeparator = requireNonNull(lineSeparator);
-        this.namedGroupFilters = ImmutableMap.copyOf(requireNonNull(namedGroupFilters));
-        this.numberedGroupFilters = ImmutableMap.copyOf(requireNonNull(numberedGroupFilters));
+        this.namedGroupFilters =
+                ImmutableMap.copyOf(requireNonNull(namedGroupFilters));
+        this.numberedGroupFilters =
+                ImmutableMap.copyOf(requireNonNull(numberedGroupFilters));
         this.maxEntries = maxEntries;
         this.keepFirst = keepFirst;
-        if (!Files.exists(file)) {
+        if (!Files.isRegularFile(file)) {
             throw new FileNotFoundException(file.toString());
         }
+
+        Matcher matcher = logPattern.matcher(defaultEntry);
+        if (!matcher.matches()) {
+            throw new IllegalArgumentException(
+                    "Default entry does not match log entry pattern");
+        }
+        try {
+            for (Map.Entry<String, Predicate<String>> f
```

Comment:
`checkFilters` short-circuits; I need to make sure all of the filters are legal with the default entry, `checkFilters` would only tell me if they all pass.

There's no error handling in `checkFilters` because by the time it is called, this block has already run, ensuring that the filters' names and indices are all valid.